mdmix/core/models.py

```diff
@@ -0,0 +1,244 @@
+                    return v * count
+                raise ValueError(f"Number of solvents ({len(v)}) must match replica count ({count})")
+        return v
+
+
+
+class SimulationRequest(BaseModel):
+    """Request model for simulation parameters"""
+    total_time: float = Field(20.0, gt=0, description="Total simulation time in ns")
+    timestep: float = Field(2.0, gt=0, description="Integration timestep in fs")
+    temperature: float = Field(300.0, gt=0, description="Temperature in K")
+    pressure: float = Field(1.0, description="Pressure in bar")
+    ensemble: EnsembleType = Field(EnsembleType.NPT, description="Thermodynamic ensemble")
+    restraint_mode: RestraintMode = Field(RestraintMode.FREE, description="Restraint mode")
+    restraint_force: float = Field(5.0, ge=0, description="Restraint force constant")
+    output_frequency: int = Field(5000, gt=0, description="Trajectory output frequency")
+    log_frequency: int = Field(1000, gt=0, description="Log output frequency")
+    engine: SimulationEngine = Field(SimulationEngine.AMBER, description="MD engine")
+
+
+class ProjectRequest(BaseModel):
+    """Complete project request model"""
+    name: str = Field(..., description="Project name")
+    description: str = Field("", description="Project description")
+    system: SystemRequest
+    replicas: ReplicaRequest
+    simulation: SimulationRequest
+    output_dir: Optional[Path] = Field(None, description="Output directory")
+    queue_system: QueueSystem = Field(QueueSystem.LOCAL, description="Queue system")
+    
+
+# Response Models (for service output)
+class SystemResponse(BaseModel):
+    """Response model for system setup results"""
+    name: str
+    status: str
+    files_created: Dict[str, Path]
+    warnings: List[str] = Field(default_factory=list)
+    errors: List[str] = Field(default_factory=list)
+
+
+class ReplicaResponse(BaseModel):
+    """Response model for replica status"""
+    name: str
+    solvent: str
+    status: str
+    directory: Optional[Path] = None
+    files: Dict[str, Path] = Field(default_factory=dict)
+    error: Optional[str] = None
+
+
+class ProjectResponse(BaseModel):
+    """Response model for complete project setup"""
+    project_name: str
+    status: str
+    created_at: datetime = Field(default_factory=datetime.now)
+    project_dir: Path
+    system: SystemResponse
+    replicas: List[ReplicaResponse]
+    total_replicas: int
+    successful_replicas: int
+    scripts_created: Dict[str, Path] = Field(default_factory=dict)
+    
+    def get_summary(self) -> Dict[str, Any]:
+        """Get project summary"""
+        return {
+            "project": self.project_name,
+            "status": self.status,
+            "total_replicas": self.total_replicas,
+            "successful": self.successful_replicas,
+            "failed": self.total_replicas - self.successful_replicas,
+            "directory": str(self.project_dir)
+        }
+
+
+class SolventInfo(BaseModel):
```

Comment:
just for consistency
```suggestion
class SolventInfoResponse(BaseModel):
```